### DISCUSSION TITLE
Adding inline nodes

### DISCUSSION THREAD

**Discussion by traed** [ueberdosis/tiptap#853] [General Questions] [answered]
Let's say I have a custom node where the top element is a span and I want to add it inside the paragraph I'm currently in. I don't want to mark text and then wrap it with something but instead insert some html at the current cursor position. Is there a way to do this? Looking at the schema for paragraph it looks like it should accept any any element with group: inline but setting this on my extension gives me an error when loading the editor: "mixing inline and block content".

**Comment by hanspagel** [comment]
Have a look here: https://github.com/ueberdosis/tiptap/issues/385#issuecomment-593330997

Hope that helps!

**Comment by traed** [comment, marked as answer]
Thanks. I was looking at that but it didn't work either. I found what was causing the error message though. Apparently you have to add `inline: true` when setting `group: 'inline'` in the schema definition, or the whole thing crashes. [This is the only documentation I found where it's used.](https://prosemirror.net/examples/schema/)

My working code if someone else needs it:
```
import {Node} from 'tiptap';
import aVueTemplate from './aVueTemplate';

export default class InlineNodeExample extends Node {

	get name() {
		return 'inline-node-example';
	}

	get schema() {
		return {
			group: 'inline',
			inline: true,
			selectable: false,
			parseDOM: [
				{
					tag: 'span'
				}
			],
			toDOM: node => [
				'span'
			]
		};
	}

	get view() {
		return aVueTemplate;
	}

	commands({type}) {
		return attrs => (state, dispatch) => {
			const {selection} = state;
			const position = selection.$cursor
				? selection.$cursor.pos
				: selection.$to.pos;
			const node = type.create(attrs);
			const transaction = state.tr.insert(position, node);

			dispatch(transaction);
		};
	}
}

**Reply by hanspagel** [reply]
Thanks for sharing!
